### ISSUE TITLE
ttnn.softmax padding support for sharded input

### ISSUE THREAD

**Issue by mtairum** [tenstorrent/tt-metal#8515]
In Mixtral8x7b we are currently making use of a padded softmax that ideally would support a sharded input.

The code snippet in `models/demos/t3000/mixtral8x7b/tt/mixtral_attention.py` looks like this:
```
attn_1B4P = attn_1B4P[:, :, :, :layer_slice]
attn_1B4P = ttnn.softmax(attn_1B4P, dim=-1)  # , memory_config=attn_output_memcfg_post_sm)
```

Since this is for decode-mode, we don't require passing an attention mask to mask other tokens, since we're only generating 1 token per iteration. As such, in the above code snippet, `layer_slice` will increase by 1 every iteration. If we don't slice it, the full tile will be account for the softmax (no mask) leading to wrong outputs.
Support for this on the op would avoid changing the memory configuration.

At the moment the tensor `attn_1B4P` is converted to interleave for the softmax and then sharded back right after. This is the current workaround. We are also working on bringing the attention mask back again and avoid doing the padded softmax, hence P2 label.
